MeshPaint/src/MeshPaintComponent.cpp

```diff
@@ -165,24 +165,24 @@ void MeshPaintComponent::paintMesh( const Ra::Engine::Rendering::Renderer::Picki
     auto& colorAttrib    = m_mesh->getCoreGeometry().getAttrib( m_currentColorAttribHdl );
     auto& colorContainer = colorAttrib.getDataWithLock();
 
-    switch ( picking.m_mode )
+    switch ( picking.getMode() )
     {
     case Ra::Engine::Rendering::Renderer::VERTEX:
         [[fallthrough]];
     case Ra::Engine::Rendering::Renderer::C_VERTEX: {
         if ( pickingRenderMode == Ra::Engine::Data::Displayable::PKM_POINTS )
         {
-            for ( auto e : picking.m_elementIdx )
+            for ( const auto& [f, v, e] : picking.getIndices() )
```

Comment:
Here (and before the other loops) we need to call `removeDuplicatedIndices` beforehand.